packages/battery/test/battery_test.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:async/async.dart';
+import 'package:flutter/services.dart';
+import 'package:test/test.dart';
+import 'package:battery/battery.dart';
+import 'package:mockito/mockito.dart';
+
+void main() {
+  MockMethodChannel methodChannel;
+  MockEventChannel eventChannel;
+  Battery battery;
+
+  setUp(() {
+    methodChannel = new MockMethodChannel();
+    eventChannel = new MockEventChannel();
+    battery = new Battery.private(methodChannel, eventChannel);
```

Comment:
I looked at the tests of the `url-launcher` plugin and compared to that I found the tests with mocking more readable. Also, the `EventChannel` doesn't seem to provide a similar mechanism for testing, so I would be back to regular mocks anyways?